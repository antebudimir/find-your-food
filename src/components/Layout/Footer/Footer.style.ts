import styled from 'styled-components';

export const FooterContainer = styled.footer`
	padding: ${(props) => props.theme.spacing.primary};
	background-color: ${(props) => props.theme.color.primary};
	text-align: center;
`;

export const FooterText = styled.p`
	color: ${(props) => props.theme.color.secondary};
`;

export const FooterLink = styled.a`
	color: ${(props) => props.theme.color.secondary};
	line-height: ${(props) => props.theme.lineHeight.primary};

	&:hover,
	&:focus {
		background-color: ${(props) => props.theme.color.secondary};
		color: ${(props) => props.theme.color.primary};
		transition: all ${(props) => props.theme.timing.primary};
	}

	@media screen and (min-width: 1024px) {
		font-size: ${(props) => props.theme.fontSize.secondary};
	}
`;
