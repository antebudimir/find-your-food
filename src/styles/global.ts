import { createGlobalStyle } from 'styled-components';
import RalewayRegular from 'fonts/Raleway-Regular.woff';
import RalewayBold from 'fonts/Raleway-Bold.woff';
import RalewayExtraBold from 'fonts/Raleway-ExtraBold.woff';

export const GlobalStyles = createGlobalStyle`
@font-face {
	font-family: 'Raleway Regular';
	src: url(${RalewayRegular}) format('woff');
	font-weight: normal;
	font-style: normal;
	font-display: swap;
}

@font-face {
	font-family: 'Raleway Bold';
	src: url(${RalewayBold}) format('woff');
	font-weight: bold;
	font-style: normal;
	font-display: swap;
}


@font-face {
    font-family: 'Raleway Extra Bold';
    src: url(${RalewayExtraBold}) format('woff');
    font-weight: bold;
    font-style: normal;
    font-display: swap;
}

* {
	outline: none;
	box-sizing: border-box;
	margin: 0;
	padding: 0;
	scroll-behavior: smooth;
}

html {
    font-family: 'Raleway Regular';
	word-wrap: break-word;
}

button,
input {
	border: none;
	display: block;
}

input,
select,
textarea,
button {
	color: inherit;
	font-size: inherit;
}

input,
select,
textarea {
	font-family: inherit;
}

h1 {
    font-family: 'Raleway Extra Bold';
}

h2,
button,
a {
	font-family: 'Raleway Bold';
}

ul,
li {
	list-style: none;
}

a {
	text-decoration: none;
}

img {
	display: block;
	max-width: 100%;
}
`;
