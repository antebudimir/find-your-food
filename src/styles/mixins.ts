import { css } from 'styled-components';

interface FlexLayoutProps {
	direction?: string;
	wrap?: string;
}

interface FlexAlignmentProps {
	justify?: string;
	align?: string;
}

export const FlexLayout = ({ direction, wrap }: FlexLayoutProps) => css`
	display: flex;
	flex-flow: ${direction} ${wrap};
`;

export const FlexAlignment = ({ justify, align }: FlexAlignmentProps) => css`
	justify-content: ${justify};
	align-items: ${align};
`;

export const hiddenVisually = css`
	position: absolute;
	clip: rect(0 0 0 0);
	clip-path: inset(50%);
	overflow: hidden;
	width: 1px;
	height: 1px;
	white-space: nowrap;
`;

export const sticky = css`
	display: flex;
	flex-flow: column nowrap;
	min-height: 89.6vh;
`;
