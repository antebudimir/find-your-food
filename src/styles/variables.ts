export const variables = {
	borderRadius: {
		primary: '0.125rem',
	},
	outline: {
		primary: '1px',
	},
	shadow: {
		primary: '#3b3f20 0px 2px 5px 0px',
		secondary: '#d45464 0px 2px 5px 0px',
		tertiary: '5px 5px 20px rgba(0, 0, 0, 0.7)',
		quaternary: '5px 5px 5px rgba(0, 0, 0, 0.7)',
	},
	sizing: {
		inputDesktopWidth: '28rem',
		summaryLinkWidth: '18.9rem', // coresponds to the width of the image
		summaryContainerWidth: '28rem',
		mainContainerWidth: '60rem',
	},
	spacing: {
		primary: '1rem',
		secondary: '0.5rem 1rem',
		tertiary: '2rem',
		quaternary: '1.5rem',
	},
	color: {
		primary: '#d45464',
		primaryLighter: '#ff6478',
		secondary: ' #ffffff',
		tertiary: '#3b3f20',
	},
	fontSize: {
		primary: '1rem',
		secondary: '1.3rem',
		tertiary: '1.5rem',
		quaternary: '2rem',
		quinary: '120%',
	},
	timing: {
		primary: '200ms',
	},
	lineHeight: {
		primary: '1.5',
		secondary: '2',
	},
};
