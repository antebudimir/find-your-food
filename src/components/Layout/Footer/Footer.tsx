import { FooterContainer, FooterLink, FooterText } from './Footer.style';

const Footer = () => {
	return (
		<FooterContainer>
			<FooterText>
				Powered by{' '}
				<FooterLink
					href="https://www.edamam.com"
					title="Go to the Edamam website"
				>
					Edamam API
				</FooterLink>
			</FooterText>
		</FooterContainer>
	);
};

export default Footer;
