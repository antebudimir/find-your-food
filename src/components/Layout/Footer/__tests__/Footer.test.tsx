import { ThemeProvider } from 'styled-components';
import { variables } from 'styles/variables';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TestRenderer from 'react-test-renderer';

import Footer from '../Footer';

test('Footer text and Footer Link render correctly', () => {
	render(
		<ThemeProvider theme={variables}>
			<Footer />
		</ThemeProvider>,
	);

	const FooterText = screen.getByText('Powered by');
	expect(FooterText).toBeInTheDocument();

	const FooterLink = screen.getByTitle('Go to the Edamam website');
	expect(FooterLink).toBeInTheDocument();
});

describe('<Footer />', () => {
	test('Footer Link navigates to root on click', () => {
		render(
			<ThemeProvider theme={variables}>
				<Footer />,
			</ThemeProvider>,
		);

		const FooterLink = screen.getByTitle('Go to the Edamam website');
		userEvent.click(FooterLink);
		expect(FooterLink.getAttribute('href')).toBe('https://www.edamam.com');
	});

	test('if the Component matches snapshot', () => {
		const component = TestRenderer.create(
			<ThemeProvider theme={variables}>
				<Footer />
			</ThemeProvider>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
