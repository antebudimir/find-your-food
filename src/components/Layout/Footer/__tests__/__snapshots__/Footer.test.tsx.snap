// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> if the Component matches snapshot 1`] = `
<footer
  className="sc-bczRLJ gpqEsY"
>
  <p
    className="sc-gsnTZi dqXVBN"
  >
    Powered by
     
    <a
      className="sc-dkzDqf cawgEO"
      href="https://www.edamam.com"
      title="Go to the Edamam website"
    >
      Edamam API
    </a>
  </p>
</footer>
`;
